lib/sqlalchemy/dialects/mysql/base.py

```diff
@@ -1766,7 +1768,21 @@ def visit_regexp_replace_op_binary(self, binary, operator, **kw):
 class MySQLDDLCompiler(compiler.DDLCompiler):
     def get_column_specification(self, column, **kw):
         """Builds column DDL."""
-
+        if (
+            self.dialect.is_mariadb is True
+            and column.computed is not None
+            and column.nullable is not True
+        ):
+            if column._user_defined_nullable is SchemaConst.NULL_UNSPECIFIED:
+                column.nullable = True
+            else:
+                raise exc.CompileError(
```

Comment:
**mike bayer** (zzzeek) wrote:

HMMMMmmmmmm are we anticipating a MariaDB error message?  We typically let the database raise these (this is a very long tradition in SQLAlchemy, there are some exceptions but overall we really try to let the DB tell us what it wants).  MariaDB server is the expert on this error message as well as if it changes someday to support this DDL, so I would remove this.

View this in Gerrit at https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/4869